### ISSUE TITLE
a11y: Add aria-labels to forms

### ISSUE THREAD

**Issue by fhemberger** [Gargron/mastodon#136]
Sorry, I don't know the right ruby/haml syntax for using attribute names including a dash, so I'm filing an issue rather than a PR:

All input fields in
```
app/views/auth/confirmations/new.html.haml
app/views/auth/passwords/new.html.haml
app/views/auth/registrations/edit.html.haml
app/views/auth/registrations/new.html.haml
app/views/auth/sessions/new.html.haml
```
are lacking proper `aria-label` attributes for screen readers. Can you please add them?
